purchase_secondary_unit/models/product.py

```diff
@@ -0,0 +1,33 @@
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import _, api, exceptions, fields, models
+from odoo.addons import decimal_precision as dp
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    uop_id = fields.Many2one(
+        comodel_name='product.uom', string='Secondary Unit of Purchase',
+        help='Specify a unit of measure here if purchasing is made in another'
+        ' unit of measure category than inventory. Keep empty to use the'
+        ' default unit of measure.')
+    uop_coeff = fields.Float(
+        string='Purchase Unit of Measure -> 2UoP Coeff',
+        digits=dp.get_precision('Product UoP'),
+        help='Coefficient to convert default Purchase Unit of Measure to'
+        ' Secondary Unit of Purchase\n uop = uom * coeff',
+        compute='_compute_uop_coeff')
+
+    @api.constrains('uop_id', 'uom_po_id')
+    def _check_uop_id(self):
+        for tmpl in self.filtered('uop_id'):
+            if tmpl.uom_po_id.category_id != tmpl.uop_id.category_id:
+                raise exceptions.ValidationError(_(
+                    'Purchase Unit of Measure and Secondary Unit of Purchase '
+                    'belong to different Category! '))
```

Comment:
if uop_id must be in same category as uom_po_id, and [Odoo checks that uom_po_id is in same category as uom_id](https://github.com/odoo/odoo/blob/3c0a0261bed4c500cf47e049e0a58c7bef616886/addons/product/models/product_template.py#L280-L284),  how can uop_id be in a different category than the inventory unit? Do I miss something?

Reply:
I'm working on it, I don't have clear how to afford some things. The uom_po_id and uop_id must be in the same category at this point, we really did not think about making them from different at this point because it uses _compute_quantity() from product.uom. In version 8 it worked making directly a conversion with uop_coeff and after watching the _compute_quantity() I have tried to adapt to this. We will give you some more info in next days.